File: bindings/php/src/ProvenAuthserver.php
<?php
// SPDX-License-Identifier: PMPL-1.0-or-later
// Copyright (c) 2026 Jonathan D.A. Jewell (hyperpolymath) <j.d.a.jewell@open.ac.uk>
//
// Auth protocol types for proven-servers.

declare(strict_types=1);

namespace ProvenServers;

/** AuthMethod matching the Idris2 ABI tags. */
enum AuthMethod: int
{
    case Password = 0;
    case Certificate = 1;
    case OAuth2 = 2;
    case Saml = 3;
    case Fido2 = 4;
    case Kerberos = 5;
    case Ldap = 6;
    case Radius = 7;
}

/** TokenType matching the Idris2 ABI tags. */
enum TokenType: int
{
    case Access = 0;
    case Refresh = 1;
    case Id = 2;
    case Api = 3;
}

/** AuthResult matching the Idris2 ABI tags. */
enum AuthResult: int
{
    case Success = 0;
    case InvalidCredentials = 1;
    case AccountLocked = 2;
    case AccountExpired = 3;
    case MfaRequired = 4;
    case IpBlocked = 5;
}

/** MfaMethod matching the Idris2 ABI tags. */
enum MfaMethod: int
{
    case Totp = 0;
    case Sms = 1;
    case Push = 2;
    case Fido2Mfa = 3;
    case Email = 4;
}

/** SessionState matching the Idris2 ABI tags. */
enum SessionState: int
{
    case Active = 0;
    case Expired = 1;
    case Revoked = 2;
    case Locked = 3;
}

File: bindings/csharp/src/ProvenAuthserver.cs
// SPDX-License-Identifier: MPL-2.0
// Copyright (c) 2026 Jonathan D.A. Jewell (hyperpolymath) <j.d.a.jewell@open.ac.uk>
//
// Auth protocol types for proven-servers.

namespace Proven;

/// <summary>AuthMethod matching the Idris2 ABI tags (0-7).</summary>
public enum AuthMethod : byte
{
    Password = 0,
    Certificate = 1,
    OAuth2 = 2,
    Saml = 3,
    Fido2 = 4,
    Kerberos = 5,
    Ldap = 6,
    Radius = 7
}

/// <summary>TokenType matching the Idris2 ABI tags (0-3).</summary>
public enum TokenType : byte
{
    Access = 0,
    Refresh = 1,
    Id = 2,
    Api = 3
}

/// <summary>AuthResult matching the Idris2 ABI tags (0-5).</summary>
public enum AuthResult : byte
{
    Success = 0,
    InvalidCredentials = 1,
    AccountLocked = 2,
    AccountExpired = 3,
    MfaRequired = 4,
    IpBlocked = 5
}

/// <summary>MfaMethod matching the Idris2 ABI tags (0-4).</summary>
public enum MfaMethod : byte
{
    Totp = 0,
    Sms = 1,
    Push = 2,
    Fido2Mfa = 3,
    Email = 4
}

/// <summary>SessionState matching the Idris2 ABI tags (0-3).</summary>
public enum SessionState : byte
{
    Active = 0,
    Expired = 1,
    Revoked = 2,
    Locked = 3
}
